File: LaboratoryWork/Semester_2/Laboratory_7/Laboratory_2/Python/task1.py
def is_prime(a: int) -> bool:
    '''Проверяет, является ли число простым.'''
    if a < 2: return False
    
    for i in range(2, int(a ** 0.5) + 1):
        if a % i == 0: return False
    
    return True


def calculateNthNum(a: int) -> int:
    '''Ищет N-й член последовательности.'''
    deviation = 1
    left_val = a
    right_val = a
    while True:
        if is_prime(left_val):
            return a - left_val
        elif is_prime(right_val):
            return a - right_val
        left_val = a - deviation   # ищем простое число слева от N
        right_val = a + deviation  # ищем простое число справа от N
        deviation += 1


def main():
    while True:  # проверка на дурака
        try:
            n = int(input("Введите свое N: "))
            break
        except ValueError:
            print("Введенное N должно быть числом!")
    
    Nth = calculateNthNum(n)
    print(f"Найден N-й член ряда: {Nth}")
    


if __name__ == "__main__":
    main()

File: LaboratoryWork/Semester_2/Laboratory_7/Laboratory_2/TypeScript/task1.ts
import * as readlineSync from 'readline-sync';

function is_prime(a: number): boolean {
    /* Проверяет, является ли число простым */
    if (a < 2) {
        return false;
    }

    for (let i: number = 2; i <= Math.floor(Math.sqrt(a)); i++) {
        if (a % i == 0) {
            return false;
        }
    }

    return true;
}


function calculateNthNum(a: number): number {
    /* Ищет N-й член последовательности. */
    let deviation: number = 1;
    let left_val: number = a;
    let right_val: number = a;

    while (true) {
        if (is_prime(left_val)) {
            return a - left_val;
        } else if (is_prime(right_val)) {
            return a - right_val;
        }
        left_val = a - deviation;
        right_val = a + deviation;
        deviation++;
    }
}


function main(): void {
    let n: number = parseInt(readlineSync.question("Введите свое N: "));

    let Nth: number = calculateNthNum(n);
    console.log("Найден N-й член ряда: " + Nth);
}


main()